### DISCUSSION TITLE
Choria in Docker + Service Agent Plugin

### DISCUSSION THREAD

**Discussion by Antiarchitect** [choria-io/general#13] [Q&A]
I'm trying to perform kinda simple thing - get all puppet service statuses (actually I want to turn off and on multiple puppet agents at once) with choria placed inside the docker container:

Dockerfile:
```
FROM centos:8

WORKDIR /

RUN curl -s https://packagecloud.io/install/repositories/choria/release/script.rpm.sh | bash && \
    yum -y update && \
    yum -y install choria nc && \
    yum -y clean all

RUN groupadd --gid 2048 choria && \
    useradd -c "Choria Orchestrator - choria.io" -m --uid 2048 --gid 2048 choria && \
    chown -R choria:choria /etc/choria && \
    mkdir /data && \
    chown choria:choria /data

USER choria
VOLUME /data

ENTRYPOINT ["/bin/choria"]
```
run.sh:
```
docker run --rm -it \
       -v "${HOME}/.choria/client.cfg:/etc/puppetlabs/mcollective/client.cfg:ro" \
       -v "${HOME}/.choria/ssl:/data/ssl:ro" \
       choria req service status service=puppet
```
And client.cfg:
```
identity = worker-1.prod.xxx
plugin.choria.middleware_hosts = puppet.xxx:4222
plugin.choria.srv_domain = puppet.xxx
plugin.choria.use_srv = false
plugin.choria.ssldir = /data/ssl
logger_type = console
loglevel = info
```

It actually does the discovery and ping well, but the command in run.sh above output is:
```
INFO[0000] Enabling support for legacy SAN free certificates  component=security ssl=file
FATA[0000] Could not run Choria: could not find DDL for agent service: could not load agent service: unknown agent service
```

I think I need to install service-agent plugin somehow but I cannot find the proper docs. Any help will be heartily appreciated :)

**Comment by ripienaar** [comment]
Yes unfortunately a fair bit more. 

The service agent is written in ruby and so we need a bunch of ruby things to make this work. 

If you look on one managed by puppet the gem installation has a mco gem installed and then a bunch of files in $libdir/mcollective/agent

**Reply by Antiarchitect** [reply]
Will installing mco gem and placing service-agent repo inside $libdir/mcollective/agent do the trick?

**Reply by ripienaar** [reply]
Probably going to need to create a shim config - see /etc/choria on a normal machine and there is a path to set. 

That will probably be it. However realise this will attempt to manage a service inside the container.

**Reply by Antiarchitect** [reply]
Thank you so much - will try!
